pennylane/drawer/tape_text.py

```diff
@@ -160,13 +168,51 @@ def _add_mid_measure_op(op, layer_str, config):
     Expectation: lambda label: f"<{label}>",
     Probability: lambda label: f"Probs[{label}]" if label else "Probs",
     Sample: lambda label: f"Sample[{label}]" if label else "Sample",
+    Counts: lambda label: f"Counts[{label}]" if label else "Counts",
     Variance: lambda label: f"Var[{label}]",
     State: lambda label: "State",
 }
 
 
+def _add_cwire_measurement_grouping_symbols(mcms, layer_str, config):
+    """Adds symbols indicating the extent of a given object for mid-circuit measurement
+    statistics."""
+    if len(mcms) > 1:
+        n_wires = len(config.wire_map)
+        mapped_bits = [config.bit_map[m] for m in mcms]
+        min_b, max_b = min(mapped_bits) + n_wires, max(mapped_bits) + n_wires
+
+        layer_str[min_b] = "╭"
+        layer_str[max_b] = "╰"
+
+        for b in range(min_b + 1, max_b):
+            layer_str[b] = "├" if b - n_wires in mapped_bits else "│"
+
+    return layer_str
+
+
+def _add_cwire_measurement(m, layer_str, config):
+    """Updates ``layer_str`` with the ``m`` measurement when it it used
+    for collecting mid-circuit measurement statistics."""
+    mcms = [v.measurements[0] for v in m.mv] if isinstance(m.mv, list) else m.mv.measurements
+    layer_str = _add_cwire_measurement_grouping_symbols(mcms, layer_str, config)
+
+    mv_label = "MCM"
+    meas_label = measurement_label_map[m.return_type](mv_label)
```

Comment:
That's a good point. However, I'd rather keep it as is because this label was approved by @trbromley . Since he's on vacation. We discussed potentially revisiting this in the future and looking into better labels for composite measurement values, so it could change later (or not) 😄

Reply:
very well